kubernetes/README.md

```diff
@@ -55,4 +55,12 @@ to create the *secret* containing the required credentials in k8s.
 - Reload metadata
   - run `hasura-dev metadata reload --endpoint "http://localhost:7070" --admin-secret HASURA_GRAPHQL_ADMIN_SECRET`
 
+## How to set environment variables
 
+1. The helm chart first takes all the variables from [values.yaml](./polkassembly/values.yaml).
+1. Then if the helm program is called with the `--values=` flag it takes all the variables from this file and applies it (possibly overwirting variables from the previous `values.yaml`).
+1. Finally, if helm is called the `--set` flags the individual variables in the `--set` will be applied to the chart. Example in `gitlab-ci.yml` [here](https://github.com/paritytech/polkassembly/blob/master/.gitlab-ci.yml#L189).
+
+Ideally, any non secret variable should be set in [values.yaml](./polkassembly/values.yaml). If the variable differs between test/prod or kusama/polkadot, it can be set in [values-dashboards-cluster-1.yaml (for test)](./polkassembly/values-dashboards-cluster-1.yaml) or [values-parity-prod.yaml (for prod)](./polkassembly/values-parity-prod.yaml).
+
+Any secret variable should be mentioned these files as well, with the value "from-gitlab-vars", and then added using the helm `--set` flags in `gitlab-ci.yml` [like here](https://github.com/paritytech/polkassembly/blob/master/.gitlab-ci.yml#L189) where it will be replaced by the variable from Gitlab, at CI runtime.
```

Comment:
```suggestion
Any secret variable should be mentioned in a `values` files as well, with any fake value such as "from-gitlab-vars", and then added using the helm `--set` flags in `gitlab-ci.yml` [like here](https://github.com/paritytech/polkassembly/blob/master/.gitlab-ci.yml#L189) where it will be replaced by the secret variable from Gitlab, at CI runtime.
```